ultralytics/models/yolo/model.py

```diff
@@ -87,18 +89,32 @@ def task_map(self):
             }
         }
 
-    def set_classes(self, classes):
+    def set_classes(self, labels, images=[]):
         """
         Set classes.
 
         Args:
-            classes (List(str)): A list of categories i.e ["person"].
+            labels (List(str)): A list of categories i.e ["person"].
+            images (List()): A list of images.
         """
-        self.model.set_classes(classes)
+
+        classes = labels.copy()
+        for i, img in enumerate(images):
+            if isinstance(img, str):
+                path = Path(img)
+                name = path.name
```

Comment:
Recommend using `path.stem` to drop the file extension when used with labeling the output image.

Reply:
I left the file extension in the label intentionally to make it easier to see what label comes from an image and what comes from a text... 
But maybe that's just the debugging perspective :)